core/config/configs.md

```diff
@@ -116,35 +116,26 @@ hand, we do know that there is a genuine need to have slightly
 in which custom behavior can be defined and recombined easily.
 Furthermore, if we always combine the methods such that the "core"
-methods are last, they cannot be overridden by the custom methods. 
-
-So we're good, right? Well, almost.
+methods are last, they cannot be overridden by the custom methods.
 
 ## Why generator functions?
 
-Sometimes, we want to burn the payment in the custom method. That
-caused a problem because with normal functions we don't actually have
-access to the full payment, the one which is built by adding methods to the
-`corePayment`. We only have access to the `corePayment` that
-is passed in as a parameter. When we call `assay.burnAll(payment)`,
-the payment gets looked up in a WeakMap of all the payments and their
-current balances. However, the corePayment won't be in that WeakMap.
-The finished payment (the corePayment plus the custom methods) is what
-is in the WeakMap. So we must have access to *that* payment. 
-
-There's a contradiction there. We wanted to ensure that the custom
-code couldn't override or mess with the core methods, but now we're
-saying that we should hand off the whole, un-`hardened` payment to the
-custom code. 
-
-There's a way to get around this contradiction: generator functions.
-We can still use the trait pattern. However, we can also give access
-to the `hardened`, final version of the payment that is the core + custom methods.
-
-Here's how that happens:
-
-* temporal dead zone
-* ??
+To use the trait pattern defensively, we must arrange for robust self-reference. In the examples above, the trait code does not refer to the object itself. But we also define traits like
+
+```js
+function* makePaymentTrait(_corePayment, assay) {
+  const payment = yield harden({
+    // This creates a new use object which destroys the payment
+    unwrap: () => makeUseObjForPayment(assay, payment),
+  });
+}
+```
+
+Above, the `payment` object is the actual payment to which this trait contributes the `unwrap` method. We introduce this peculiar generator pattern to solve this self-reference problem. The `payment` variable refers to the object as a whole. However, the object-as-a-whole is not yet assembled, and will be assembled only by a distinct piece of code written in a distinct scope.
+
+In the generator function pattern shown above, the trait methods of a given layer are defined in the scope of the `payment` variableis bound on the left of the yield. The `payment` variable is bound to the value that the yield returns. However, before that happens, the generator yields the argument to yield, the object containing the trait's methods, to be combined to make the object itself. The pattern above containing `makeMintTraitIter` first runs one step of the trait generartor to obtain the trait methods, combining them into the overall object. Only when the object is complete does it go back to the generator, in order to bind that object to that trait's variable for the object itself, such as `payment`.
```

Comment:
something is broken in the vicinity of "varialbleis", and I can't tell what it should be.

Reply:
Done